src/instruments/src/PFD/LandingSystemIndicator.tsx

```diff
@@ -42,19 +50,37 @@ export class LandingSystem extends DisplayComponent<{ bus: EventBus, instrument:
         sub.on(getDisplayIndex() === 1 ? 'ls1Button' : 'ls2Button').whenChanged().handle((lsButton) => {
             this.lsButtonPressedVisibility = lsButton;
             this.lsGroupRef.instance.style.display = this.lsButtonPressedVisibility ? 'inline' : 'none';
+            this.deviationGroup.instance.style.display = this.lsButtonPressedVisibility ? 'none' : 'inline';
             this.handleGsReferenceLine();
         });
 
         sub.on('altitudeAr').handle((altitude) => {
             this.altitude = altitude;
             this.handleGsReferenceLine();
         });
+
+        sub.on(getDisplayIndex() === 1 ? 'ldevRequestLeft' : 'ldevRequestRight').whenChanged().handle((ldevRequest) => {
+            this.ldevRequest = ldevRequest;
+            this.updateLdevVisibility();
+        });
+
+        sub.on('xtk').whenChanged().handle((xtk) => {
+            const xtkValid = Math.abs(xtk) > 0;
+            if (xtkValid !== this.xtkValid) {
```

Comment:
You can just use a Subject here and sub on it to update the visibility, in onAfterRender.